### DISCUSSION TITLE
Is GCC's `-fstrict-aliasing -fno-ipa-strict-aliasing` sufficient to avoid possible miscompilation?

### DISCUSSION THREAD

**Discussion by johnsonjh** [libuv/libuv#4637] [Q&A]
For an application (built using GCC with LTO) is it sufficient to build the application with `-fstrict-aliasing -fno-ipa-strict-aliasing` to avoid possible miscompilations (where libuv itself is built with `-flto -fno-strict-aliasing` and usually linked statically)?

Currently, we always build both libuv and the application with `-fno-strict-aliasing` for safety, but our benchmarks indicate that `-fstrict-aliasing` results in a measurable 3.5% performance improvement, and `-fstrict-aliasing -fno-ipa-strict-aliasing` has very similar results.

For Clang, it seems there is no `-fno-ipa-strict-aliasing` flag currently.

**Comment by bnoordhuis** [comment]
Inactionable as a bug report, converted to a discussion. I don't know the answer, by the way, not with enough confidence to say for sure.
